cli_tools/google-osconfig-agent/config/config_test.go

```diff
@@ -76,3 +76,56 @@ func TestSetConfig(t *testing.T) {
 		t.Errorf("Default poll interval: got(%f) != want(%d)", SvcPollInterval().Minutes(), 3)
 	}
 }
+
+func TestSetConfigDefaultValues(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `{"instance":{"id":12345,"name":"name","zone":"zone","attributes":{"os-config-endpoint":"SvcEndpoint"}}}`)
+	}))
+	defer ts.Close()
+
+	if err := os.Setenv("GCE_METADATA_HOST", strings.Trim(ts.URL, "http://")); err != nil {
+		t.Fatalf("Error running os.Setenv: %v", err)
+	}
+
+	if err := SetConfig(); err != nil {
+		t.Fatalf("Error running SetConfig: %v", err)
+	}
+
+	testsString := []struct {
+		desc string
+		op   func() string
+		want string
+	}{
+		{"SvcEndpoint", SvcEndpoint, "SvcEndpoint"},
+		{"Instance", Instance, "zone/instances/name"},
+		{"ID", ID, "12345"},
+		{"ProjectID", ProjectID, "projectId"},
+		{"Zone", Zone, "zone"},
+		{"Name", Name, "name"},
```

Comment:
Is this testing these defaults? It looks like you set them on line 82.

Reply:
no.. i only intended to test the metadata attributes added by end end user since those may or may not present These are infra level attributes and i assumed that these will be already during events like project setup, instance spin up

Reply:
Should they be removed from this test then? It looks like they're already tested above.

Reply:
i actually updated our tests to test for default svcendpoint

Reply:
Oh. I see. LGTM 👍